CHANGELOG.md

```diff
@@ -26,6 +26,8 @@
 - Implement user-facing bech32 encoding for `AccountId`s (#1185).
 - [BREAKING] Added `InitStorageData::from_toml()`, improved storage entry validations in `AccountComponentMetadata` (#1170).
 - [BREAKING] Rework miden-lib error codes into categories (#1196).
+- Added `info` log level by default in the proving service (#1200).
+- [BREAKING] Added port check in the proxy service to avoid starting up if the port is in use (#1200).
```

Comment:
I don't believe this truly addresses the issue. Though I do think this is a good check to do since I don't believe we want the upgrade behaviour - so its good to check whether we are running a duplex service.

Unless you're suggesting there was another proxy running the entire time, which held onto that port? I don't think that's what happened, because it started working once we changed the worker port (and not the proxy's).

When the proxy connects to a worker on port `N` then the proxy itself should be using a random local port to create that connection. In other words the local port `N` should still be available for use for serving.

We _should_ be able to run all workers on the same port `N` on separate instances, and _also_ serve the proxy gRPC on port `N`.